modules/Rock.cmake

```diff
@@ -1352,6 +1369,35 @@ function(rock_gtest TARGET_NAME)
     rock_add_test(${TARGET_NAME} "${__rock_test_parameters}")
 endfunction()
 
+function(rock_coverage_report_generation TARGET_NAME)
+    message(STATUS "Generating report for ${TARGET_NAME} package")
+    find_program(
+        gcovr_exec NAMES
+        ${ROCK_GCOVR_EXECUTABLE} gcovr
+    )
+    if (NOT gcovr_exec)
+        message(FATAL_ERROR "Could not find an executable for gcovr.")
+    endif()
+
+    set(gcovr_config_option --filter ${PROJECT_SOURCE_DIR}/src/ --html -o ${PROJECT_SOURCE_DIR}/coverage.html)
+    if(${ROCK_COVERAGE_REPORT_PATH})
+        message(warning "Setting an explicit config path coverage report")
+        set(gcovr_config_option --filter ${PROJECT_SOURCE_DIR}/src/ --html -o ${ROCK_COVERAGE_REPORT_PATH}/coverage.html)
+    endif()
+
+    if(ROCK_TEST_CXX_SONARQUBE_COVERAGE_ENABLED AND ROCK_TEST_CXX_GCOVR_GENERATION_ENABLED)
+        list(APPEND gcovr_config_option --sonarqube -o sonarcube_coverage_report)
+    endif()
+
+    add_test(
+        NAME
+        report_generation
+        COMMAND
+        ${gcovr_exec}
+        ${gcovr_config_option}
+    )
```

Comment:
You need to add a dependency on this target ("report_generation") to the corresponding tests, or - rather - let the  caller pass them to you.

Also, TARGET_NAME is meant to be the name of the test target (NAME ${TARGET_NAME}) so that one can generate more than one report if needed.

Reply:
The report is a .html with the coverage information for all archives from the /src folder, is a target to generate more than one report really necessary? I don't understand the benefits, could you expand a bit?